react-js/styling-react.md

```diff
@@ -0,0 +1,252 @@
+		{/* End of Parent Div */}  
+	</div>
+      ```
+2. Using External CSS 
+    1. Create an external CSS file and add a parent class with same styling as you did in inline CSS.
+	2. Create a class for each **child element**.
+	3. Your CSS file will look like -
+	- ```
+      .parent {
+		margin: 10px 0;
+		display: flex;
+		width: 100vw;
+		background-color: darkorange;
+	  }
+			
+	  .child1 {
+		flex-basis: 25%;
+		max-width: 25%;
+		padding: 10px;
+		background-color: green;
+	  }
+			
+	  .child2 {
+		flex-basis: 25%;
+		max-width: 25%;
+		padding: 10px;
+		background-color: red;
+	  }
+
+	  .child3 {
+		flex-basis: 25%;
+		max-width: 25%;
+		padding: 10px;
+		background-color: blue;
+	  }
+      	```
+	#### Note - CSS Selectors can be used as an alternative, instead of providing different classes to each child element. Background color can be added in different class to child *div* elements.
+	- For Example - 
+	```
+   .parent div {
+	  flex-basis: 25%;
+	  max-width: 25%;
+	  padding: 10px;
+	}
+
+	.redColor {
+	  background-color: red;
+	}
+
+	.greenColor {
+	  background-color: green;
+	}
+
+	.blueColor {
+	  background-color: blue;
+	}
+
+	```
+	4. Now we import this external file in ``App.js`` and add classes to **div**.
+	 - > ``` import './external.css```
+	-  > ``` <div className="parent"> </div> ```
+	-  > ``` <div className="child1">Child Div 1 </div> ```
+	   - Finally your `App.js` will look like 
+	```
+	import React from 'react';
+	import './App.css';
+	import './external.css'
+	function App() {
+  	  return (
+        <div className="App">
+		  {/* Start of Parent Div */}
+      		<div className="parent">
+			  {/* Start of Child Divs */}
+			  <div className="child1">Child Div 1</div>
+			  <div className="child2" >Child Div 2</div>
+			  <div className="child3">Child Div 3</div>
+			  {/* End of Child Divs */}
+	  		</div>
+			{/* End of Parent Div */}  
+		</div>
+  		);
+	}
+
+	export default App;
+
+	```
+
+3. Using Bootstrap
+	1. Let's add Bootstrap CDN in index.html file.
+	2. Create a row and make child **div** as `col-3`.
+	3. Since Bootstrap has different colors, we have to add `backgroundColor` as inline CSS to differentiate between child element **div**.
+	4. `App.js` will look like -
+    - ```
+	  <div className="App">
+		{/* Start of Parent Div */}	
+          <div className="row mt-2 mb-2" style={{backgroundColor:"darkorange"}}>
+		    {/* Start of Child Divs */}
+			 <div className="col-3 p-2" style={{backgroundColor:"green"}}>Child Div 1</div>
+			 <div className="col-3 p-2" style={{backgroundColor:"red"}}>Child Div 2</div>
+			 <div className="col-3 p-2" style={{backgroundColor:"blue"}}>Child Div 3</div>
+			 {/* End of Child Divs */}
+		</div>
+		{/* End of Parent Div */}
+	  </div>
+      ```
+
+
+### Common Mistakes and their Correction
+
+- Combining different classes based on certain state/props.
+  - When a component is receiving a prop and depending upon the value of that prop the CSS is rendered.
```

Comment:
Can you make it clear whether the following example is a mistake, or the correct way to do it?  You've already made it clear for "Wrong path" and "Camelcase", but not this one.

Reply:
Just this and transferring the slides, and you’re done ✅

Reply:
I changed the label to hard, as I appreciate how much work you put into the slides.

Reply:
Thank you @alodahl , I have transferred the ownership of slides and also covered this in the new commit.